PowerUp/app/src/main/java/powerup/systers/com/AvatarRoomActivity.java

```diff
@@ -65,13 +65,33 @@ public void onCreate(Bundle savedInstanceState) {
             SessionHistory.hasPreviouslyCustomized = true;
         } else {
             // if yes, check the customization from database for eyes, skin, hair, skin
-            eye=getmDbHandler().getAvatarEye();
-            skin=getmDbHandler().getAvatarSkin();
-            hair=getmDbHandler().getAvatarHair();
-            cloth=getmDbHandler().getAvatarCloth();
+            source.getAvatarEye(new IDataSource.LoadIntegerCallback() {
+                @Override
+                public void onResultLoaded(int value) {
+                    eye = value;
+                }
+            });
+            source.getAvatarSkin(new IDataSource.LoadIntegerCallback() {
```

Comment:
@manjumegh @niksj @manishakeim I need reviews here, is it a good approach? As i'm getting value inside callback & I need to assign the value to a variable to use further in class.

Reply:
Yes, I don't see any problem with this.

Reply:
I am not sure, if this works. Looks like you are having async callback, where you are setting eye, skin, hair etc some value. It may be possible in some scenario, if the values are not set as soon as function call, as this is async call. and later in the code, you are using eye, skin, hair without checking valid value. 
either you have sync callback, where you wait till it get assign, if it get stuck due to any reason. 
or you check before value assignment and log appropriate msg/action.